### DISCUSSION TITLE
【SPAM対策アイデア】IPアドレス形式URLの禁止、リンクだけの投稿禁止

### DISCUSSION THREAD

**Discussion by shiracamus** [increments/qiita-discussions#583] [要望・アイデア・不具合]
スパムの中には、IPアドレス形式URLが書いてありどこの国のどのようなホストなのか分からず、広告や危険なサイトに誘導されてしまうこともあるので投稿禁止にしてはいかがでしょうか。プロフィールのURLも同様に。「不適切なURLが含まれています」というダイヤログを表示するのもいいかと思います。
また、URLだけ書いてある記事（概要説明などがない）は、Qiitaの趣旨である「自分の知識を共有する」というポリシーから外れていると思うので、投稿禁止してはいかがでしょうか。Qiitaガイドラインを提示して「対象環境や説明を書きましょう」と促すダイヤログを表示するのもいいかと思います。
検討していただけたら幸いです。

**Comment by shiracamus** [comment]
スパムではないですが、コードブロックだけを投稿して何の説明もない記事も散見されるので、Qiitaガイドラインに書いてある「環境や前提条件を説明する」を促すように「動作環境や前提知識、記事概要を書きましょう」とダイヤログを表示して、記事の価値を高めることに繋げてはいかがでしょうか。

**Comment by tanaka-asami415** [comment]
@shiracamus 
アイデアありがとうございます🙏

スパム対策のため詳細をお伝えすることはできないのですが、少しでも改善ができるよう継続的な修正を行っております。
今回頂いた内容も、今後の改善の参考にさせていただきます🙇

**Comment by juner** [comment]
@shiracamus 最近だと タイトルが URL だけのタイプのスパム記事も増えてきてて 心配になりつつある （後日確認すると そのタイトルのURLが有効になっているのを確認できるので、遅延性の宣伝な可能性があります。

![image](https://github.com/user-attachments/assets/9198dfe6-8e3e-4aaf-85dd-30309c66ac47)

記事内容を調べてみたところ、 AI ではなくて、他の記事や投稿を組み合わせた謎の記事ということがわかりましたが、目的が不明です。

**Reply by shiracamus** [reply]
この3連休で500記事くらい投稿されそうな勢いです。
SPAM報告してましたがあまりにも多いため報告をやめます。
SPAMフィルターでの自動対処を急いだ方がいいです。
https://qiita.com/search?q=title%3Ahttps&sort=created

**Reply by juner** [reply]
結構致命的な検索汚染になるSPAMな気もします。（通報も結構手間なので。

**Comment by devtolog** [comment]
URLを開いて翻訳するとあからさまなサポート詐欺（？）のようなものでした。電話番号みたいなものはクリックしたら危なそうです。そもそも、本当の航空会社が払い戻しをするのであれば、Qiitaに投稿する意味がありません。また、全て`https://rb.gy/~`のようになっています。

2025/2/22~2025/2/25までの記事数：560記事

**Comment by devtolog** [comment]
アカウントが違うのに本文の内容が全く同じなので、多分プログラミングで記事が投稿されてるかも。

**Comment by ko1nksm** [comment]
詐欺でしょうね。検索すると海外のサイトでもいくつか被害にあっているのを見かけます。

ちなみに `https://rb.gy/~` 以外にもいくつかあります。 `https://dub.sh` なんたらとか。
短縮 URL は URL の先を確認するなどの対策が必要かと思われます。

本格的なスパム対策の前に、仮で簡単な対策を行ったほうが良いでしょう。
例えばスパムと見られる投稿やアカウントは、一定期間、公開を遅らせるとかですね。特に土日は。
本人以外には表示されないような対策も考えられます。

新規アカウントは、ユーザーからのスパム報告が多ければ自動的に非公開する機能など、
難しい対策の前にやれることはあるはずです。

**Reply by ko1nksm** [reply]
昼か夕方ぐらいに確認した所、スパム投稿は削除されていたようですが再度投稿されているようです。

- https://qiita.com/arienejacob954
- https://qiita.com/augustchjepu

試しに `https://rb.gy/` を NG ワードにしてみてみるのはいかがでしょうか？
おそらく別の短縮 URL を使ってくることが想定されますが、スパムに使われやすい短縮 URL サービスを調べる手間が省けます。

**追記** さらに増えていました。

- https://qiita.com/dorile
- https://qiita.com/wevano
- https://qiita.com/nesabeg

夜間や休日など、しばらく対応されない時間を狙っているように思えます。
スパム対応を24時間手動で行うのは現実的ではないため
管理者がいなくともある程度対応可能な何かしらの手段を取り入れたほうが良いでしょう。

**Comment by h-kono-it** [comment]
スパム報告ベースでの自動非公開には慎重な立場です。
複数アカウントを作って新着記事を通報して嫌がらせを行う等の可能性が排除しきれないためとなります。

代案のないコメントとなることをお許しください。

**Reply by ko1nksm** [reply]
本当にそうでしょうか？ 例えばスパム報告できるユーザーを一定期間Qiitaの利用者であり一定数の記事を投稿した人などの条件をつければ可能になりそうに思えます。これなら嫌がらせを行うなどの可能性を排除できますよね？ 試す価値は十分あると思います。

**Reply by h-kono-it** [reply]
対応に合理性も実現性もあると思います。
それでもなお、私は慎重な立場を取ります。

文面から次の2つのパターンが考えられると思います。
1. 一定の利用実績に伴いスパム報告を解禁する
2. 一定の利用実績に伴いスパム報告の意味に差をつける

どちらにせよ、1ユーザーに対して「ランク」のようなものが付与され特権的に扱われるように思えます。（それが外からわかるかは別として）

私は運営者ではないので1ユーザーとしての意見ですが、プラットフォームとしてユーザーアカウントをランク付けするようなものは好みではありません。
あくまで「好みではない」レベルなので、「反対」というわけではありません。

ご提案いただきありがとうございます。

**Reply by juner** [reply]
仲の悪いユーザもいるので 特権により 変に悪用されるのはまずい わかります。

**Reply by ko1nksm** [reply]
特権と言い換えないでください。通報は現在でも誰でもできる権利です。ただ「悪用されるから」一定期間無効なだけです。
仲が悪いならブロック機能がありますし、悪用されたのであればその後に考えれば良いことです。
存在しない問題を持ち出してきて「問題があるからやめよう」と言うのではなく、問題が起きてから考えましょう。

追記: 一人のユーザーの通報によって非表示するという話ではないので「仲が悪いユーザー」は問題にならないでしょうし、記事の内容がスパムではないのに組織的な通報があれば一度目は上手く言ったとしてもすぐにばれるでしょう。

**Comment by tanaka-asami415** [comment]
ご報告が遅くなり申し訳ありません。この度はご不便をおかけしました🙇
一次対応として、直近増えているスパムに対して、自動検知を行えるように改善を行いました。

通報のご協力、ありがとうございます。
いただいている内容も検討しながら、引き続き対策を行っていきます🙇

**Reply by ko1nksm** [reply]
対応ありがとうございます。

次は「www.rb.gy/～」のような形のURLに変更したようです。（一例）

- https://qiita.com/mrtibbless
- https://qiita.com/flygon

以下の実験から今度はコメントに対してスパムを行うことが想定されます。

- https://qiita.com/sholom/items/76fdb1dbb2b9f5c0cc7e

正直、日本のサイトにこのようなスパムを行っても効果は限りなくゼロのはずですが、おそらくスパム投稿が簡単に行えるサイトだとスパマーに目をつけられているのではないかと考えられます。

**Reply by tanaka-asami415** [reply]
ご共有ありがとうございます。
再度お手数をおかけしてしまい申し訳ありません🙇

ご指摘のURL形式についても追加で対応を進めます。
抜本的な対策についても現在進めておりますが、実現までには時間がかかる見込みです。
それまでは暫定的な対応となりますが、引き続き対策を実施していきます。

ご不便をおかけいたしますが、引き続きよろしくお願いします。
